src/collection-facets/facet-row.ts

```diff
@@ -0,0 +1,271 @@
+            title=${onlyShowText}
+            id=${showOnlyCheckboxId}
+          />
+          <input
+            type="checkbox"
+            id=${negativeCheckboxId}
+            .name=${facetType}
+            .value=${bucket.key}
+            @click=${(e: Event) => {
+              this.facetClicked(e, true);
+            }}
+            .checked=${facetHidden}
+            class="hide-facet-checkbox"
+          />
+          <label
+            for=${negativeCheckboxId}
+            class="hide-facet-icon${facetHidden ? ' active' : ''}"
+            title=${showHideText}
+          >
+            <span class="eye">${eyeIcon}</span>
+            <span class="eye-closed">${eyeClosedIcon}</span>
+          </label>
+        </div>
+        <label
+          for=${showOnlyCheckboxId}
+          class="facet-info-display"
+          title=${onlyShowText}
+          aria-label=${ariaLabel}
+        >
+          <div class="facet-title">${bucketTextDisplay}</div>
+          <div class="facet-count">${bucket.count.toLocaleString()}</div>
+        </label>
+      </div>
+    `;
+  }
+
+  //
+  // EVENT HANDLERS & DISPATCHERS
+  //
+
+  /**
+   * Handler for whenever this facet is clicked & its state changes
+   */
+  private facetClicked(e: Event, negative: boolean) {
+    const { bucket, facetType } = this;
+    if (!bucket || !facetType) return;
+
+    const target = e.target as HTMLInputElement;
+    const { checked } = target;
+    bucket.state = FacetRow.getFacetState(checked, negative);
+
+    this.dispatchFacetClickEvent({
+      facetType,
+      bucket,
+      negative,
+    });
+  }
+
+  /**
+   * Emits a `facetClick` event with details about this facet & its current state
+   */
+  private dispatchFacetClickEvent(detail: FacetEventDetails) {
+    const event = new CustomEvent<FacetEventDetails>('facetClick', {
+      detail,
+    });
+    this.dispatchEvent(event);
+  }
+
+  //
+  // OTHER METHODS
+  //
+
+  /**
+   * Returns the composed facet state corresponding to a positive or negative facet's checked state
+   */
+  static getFacetState(checked: boolean, negative: boolean): FacetState {
+    let state: FacetState;
+    if (checked) {
+      state = negative ? 'hidden' : 'selected';
+    } else {
+      state = 'none';
+    }
+    return state;
+  }
+
+  //
+  // STYLES
+  //
+
+  static get styles(): CSSResultGroup {
+    return css`
+      async-collection-name {
+        display: contents;
+      }
+      .facet-checkboxes {
+        margin: 0 5px 0 0;
+        display: flex;
+        height: 15px;
+      }
+      .facet-checkboxes input:first-child {
+        margin-right: 5px;
+      }
+      .facet-checkboxes input {
+        height: 15px;
+        width: 15px;
+        margin: 0;
+      }
+      .facet-row-container {
+        display: flex;
+        font-weight: 500;
+        font-size: 1.2rem;
+        margin: 2.5px auto;
+        height: auto;
+        border-top: var(--facet-row-border-top, 1px solid transparent);
```

Comment:
I would love to defined these as const just above tthe `return css` statement where a
ll the css variable can placed gives more readability.

`const facetRowBorderTop = css`var(--facet-row-border-top, 1px solid transparent);``

and use this const in css
 `border-bottom: ${facetRowBorderTop}`